WordPress/src/main/java/org/wordpress/android/ui/mediapicker/loader/GifMediaDataSource.kt

```diff
@@ -0,0 +1,131 @@
+package org.wordpress.android.ui.mediapicker.loader
+
+import android.content.Context
+import android.net.Uri
+import com.tenor.android.core.constant.AspectRatioRange
+import com.tenor.android.core.constant.MediaCollectionFormat
+import com.tenor.android.core.constant.MediaFilter
+import com.tenor.android.core.model.impl.Result
+import com.tenor.android.core.network.ApiClient
+import com.tenor.android.core.network.IApiClient
+import com.tenor.android.core.response.impl.GifsResponse
+import org.wordpress.android.R
+
+import org.wordpress.android.ui.mediapicker.MediaItem
+import org.wordpress.android.ui.mediapicker.MediaItem.Identifier.GifMediaIdentifier
+import org.wordpress.android.ui.mediapicker.MediaType.IMAGE
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult.Failure
+import org.wordpress.android.ui.mediapicker.loader.MediaSource.MediaLoadingResult.Success
+import org.wordpress.android.viewmodel.gif.provider.GifProvider.GifRequestFailedException
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class GifMediaDataSource
+@Inject constructor(
+    private val context: Context,
+    private val tenorClient: IApiClient
+) : MediaSource {
+    private var nextPosition: Int = 0
+    private var items = mutableListOf<MediaItem>()
+    private var lastFilter: String? = null
+
+    override suspend fun load(forced: Boolean, loadMore: Boolean, filter: String?): MediaLoadingResult {
+        if (lastFilter != filter) {
```

Comment:
I was thinking about this approach (I know I used it in the stock data source). I think it's enough to check here whether the `loadMore == false`. We want to clear or the previous data every time when we are not loading the next page. Does it make sense?

Reply:
Since tenor seems to have finite streams of items per each filter string, I was thinking this to be more robust in the edge case when you reach the end of list, but tbh I just tried the edge case and it works as expected, so happy to change it 👍  (changed in 7a5d7f94cc965e6135ae82d67391bf8d96a46c1b)